### PR TITLE
[CSS in JS] BaseIcon in utils/icon

### PR DESCRIPTION
**Pull request by grosbouddha**

### CODE REVIEW

**Comment by grosbouddha** [comment]
Tested locally: loader, all icons, a few random icons

File: src/_utils/icon/index.tsx [hunk 1]
```diff
@@ -1,63 +1,28 @@
...
+`
 
-BaseIcon.defaultProps = {
-  className: '',
-  iconColor: color.icon,
-  size: 24,
-  title: '',
-  badgeAriaLabel: '',
-  badgeContent: '',
+StyledBaseIcon.defaultProps = {
```
Comment (PerrineBoissieres) [inline src/_utils/icon/index.tsx]: Keep the defaultProps on the BaseIcon component, no need add them on the styled one.

Comment (grosbouddha) [reply src/_utils/icon/index.tsx]: Done, PTAL

File: src/typings/icons.d.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 declare interface Icon {
   readonly size?: number | string
   readonly className?: Classcat.Class
+  readonly extraClassName?: Classcat.Class
```
Comment (Tuxosaurus) [inline src/typings/icons.d.ts]: to be removed after the newest discoveries

**Pull request merged**